src/System.Drawing.Common/tests/Imaging/EncoderParameterTests.cs

```diff
@@ -0,0 +1,319 @@
+                yield return new object[] { new Action(() => new EncoderParameter(null, (short)0)) };
+                yield return new object[] { new Action(() => new EncoderParameter(null, numerator: 0, denominator: 0)) };
+                yield return new object[] { new Action(() => new EncoderParameter(null, rangebegin: 0, rangeend: 0)) };
+                yield return new object[] { new Action(() => new EncoderParameter(null, 0, 0, 0, 0)) };
+                yield return new object[] { new Action(() => new EncoderParameter(null, "anyString")) };
+                yield return new object[] { new Action(() => new EncoderParameter(null, new byte[] { })) };
+                yield return new object[] { new Action(() => new EncoderParameter(null, new byte[] { }, false)) };
+                yield return new object[] { new Action(() => new EncoderParameter(null, new short[] { })) };
+                yield return new object[] { new Action(() => new EncoderParameter(null, new long[] { })) };
+                yield return new object[] { new Action(() => new EncoderParameter(null, new int[] { }, new int[] { })) };
+                yield return new object[] { new Action(() => new EncoderParameter(null, new long[] { }, new long[] { })) };
+            }
+        }
+
+        [ConditionalTheory(nameof(PlatformDetection) + "." + nameof(PlatformDetection.IsNotWindowsNanoServer))]
+        [MemberData(nameof(EncoderParameterCtor_NullEncoder_TestData))]
+        public void Ctor_NullEncoder_ThrowsNullReferenceException(Action encoderCtor)
+        {
+            Assert.Throws<NullReferenceException>(encoderCtor);
+        }
+
+        [ConditionalTheory(nameof(PlatformDetection) + "." + nameof(PlatformDetection.IsNotWindowsNanoServer))]
+        [InlineData(EncoderParameterValueType.ValueTypeByte, int.MaxValue, typeof(AccessViolationException))]
+        [InlineData(EncoderParameterValueType.ValueTypeAscii, int.MaxValue, typeof(AccessViolationException))]
+        [InlineData(EncoderParameterValueType.ValueTypeShort, (int.MaxValue / 2) + 1, typeof(OverflowException))]
+        [InlineData(EncoderParameterValueType.ValueTypeLong, (int.MaxValue / 4) + 1, typeof(OverflowException))]
+        [InlineData(EncoderParameterValueType.ValueTypeRational, (int.MaxValue / 8) + 1, typeof(OverflowException))]
+        [InlineData(EncoderParameterValueType.ValueTypeLongRange, (int.MaxValue / 8) + 1, typeof(OverflowException))]
+        [InlineData(EncoderParameterValueType.ValueTypeUndefined, int.MaxValue, typeof(AccessViolationException))]
+        [InlineData(EncoderParameterValueType.ValueTypeRationalRange, (int.MaxValue / 16) + 1, typeof(OverflowException))]
+        public void Ctor_Encoder_TooBigNumberOfValues_Type_Value_AccessViolationException(EncoderParameterValueType type, int numberOfValues, Type expected)
+        {
+            IntPtr anyValue = IntPtr.Zero;
```

Comment:
Nit: if you're changing this file for PR feedback you might as well inline this value